### PR TITLE
Updated the flow control tutorial

### PR DESCRIPTION
**Pull request by evedon**
Some updates:
- Corrected several inaccuracies in tutorial.
- Busy wait is only mentionned at the beginning ; it is emphasized that if possible alternatives should be used.
- Introduced `ThisThread::sleep_for` and `thread_sleep_for` APIs.
- Renamed `Busy wait button` section to `Active polling button`.

### CODE REVIEW

**Comment by AnotherButler** [comment]
I think this will pass Travis again if we rebase.

**Comment by AnotherButler** [comment]
@iriark01 This is waiting on engineering review. It will then be ready for review.

**Review by bulislaw — APPROVED** [review]

File: docs/api/io/flow_control.md [hunk 1]
```diff
@@ -6,56 +6,64 @@ We can use Blinky to explore flow control and task management in Arm Mbed OS app
...
+
+Busy wait is a method that blocks the processor for a period of time. This is an effective way to create time delays, but it is inefficient because it wastes processor time and keeps the processor running at full power for the duration of the wait:
 
 [![View code](https://www.mbed.com/embed/?url=https://github.com/ARMmbed/mbed-os-examples-docs_only/blob/master/Tutorials_UsingAPIs/Flow-Control-Busy-Wait)](https://github.com/ARMmbed/mbed-os-examples-docs_only/blob/master/Tutorials_UsingAPIs/Flow-Control-Busy-Wait/main.cpp)
 
 Notice `printf()`; you can enable this by uncommenting the line (remove the `//`). `printf()` prints to the terminal, so you can use it to get debug information.
 
-<span class="tips">**Tip:** We recommend using [CoolTerm](http://freeware.the-meiers.org/), as it works the same on Windows, Linux and OS X. [Here is a handy video on how to use CoolTerm](https://www.youtube.com/watch?v=jAMTXK9HjfU) to connect to your board and view the `printf()` statements. For more information, see our [serial communication tutorials](../tutorials/serial-communication.html).</span>
+<span class="tips">**Tip:** Busy wait is inefficient and should be avoided. </span>
```
Comment (iriark01) [inline docs/api/io/flow_control.md]:
```suggestion
<span class="tips">**Tip:** Busy wait is inefficient, and we recommend not using it. </span>
```

File: docs/api/io/flow_control.md [hunk 2]
```diff
@@ -6,56 +6,64 @@ We can use Blinky to explore flow control and task management in Arm Mbed OS app
...
+
+#### Suspend thread execution
+
+A better alternative to busy wait is to suspend thread execution while waiting. You can use the `ThisThread::sleep_for` C++ API or `thread_sleep_for` C API.
```
Comment (iriark01) [inline docs/api/io/flow_control.md]:
```suggestion
A better alternative to busy wait is to suspend thread execution while waiting. Use:

 * For C++: The `ThisThread::sleep_for` API 
 * For C: The `thread_sleep_for` API.
```

File: docs/api/io/flow_control.md [hunk 3]
```diff
@@ -6,56 +6,64 @@ We can use Blinky to explore flow control and task management in Arm Mbed OS app
...
 
 ### Ticker
 
-Tickers and timers are another way of creating a time interval. These methods are somewhat better than busy wait because they allow other code to run while you are waiting. It is even possible, though nontrivial, to sleep during the wait period.
+Tickers and timers are another way of creating a time interval. These methods allow other code to run while you are waiting. During the wait period, if no threads are running, the idle thread can automatically put the system to sleep.
 
-Here is an example that doesn't include sleeping:
+If you don't need the precision of a high-frequency timer/ticker, we recommend that you use LowPowerTimer/LowPowerTicker instead because this will allow the system to be put in deep sleep mode.
```
Comment (iriark01) [inline docs/api/io/flow_control.md]:
```suggestion
If you don't need the precision of a high-frequency timer or ticker, we recommend that you use LowPowerTimer or LowPowerTicker instead. This allows the system to be put in deep sleep mode.
```

**Review by iriark01 — CHANGES_REQUESTED** [review]

**Comment by iriark01** [comment]
Should this go out to 6.0-alpha as well?

**Comment by evedon** [comment]
> Should this go out to 6.0-alpha as well?

yes

**Comment by iriark01** [comment]
Merging; will cherry pick into alpha

**Pull request merged**